modules/component/src/core/render-scheduler.ts

```diff
@@ -0,0 +1,32 @@
+import { ChangeDetectorRef, NgZone } from '@angular/core';
+
+export interface RenderScheduler {
+  schedule(): void;
+}
+
+export interface RenderSchedulerConfig {
+  ngZone: NgZone;
+  cdRef: ChangeDetectorRef;
+}
+
+export function createRenderScheduler(
+  config: RenderSchedulerConfig
+): RenderScheduler {
+  function schedule(): void {
+    if (hasZone(config.ngZone)) {
+      config.cdRef.markForCheck();
+    } else {
+      config.cdRef.detectChanges();
+    }
+  }
+
+  return { schedule };
```

Comment:
Why do we wrap `schedule` in an object here?

Reply:
Good point! Creating `RenderScheduler` as object will make it flexible for further improvements. E.g. if we decide to add `local` and `global` strategy, `RenderScheduler` will contain `flush` method. Something like this:

```ts
export interface RenderScheduler {
  schedule(): void;
  flush(): void;
}

export function createRenderScheduler(config: RenderSchedulerConfig): RenderScheduler {
  return config.strategy === RenderStrategy.Local
    ? createLocalRenderScheduler(config)
    : createGlobalRenderScheduler(config);
}

function createLocalRenderScheduler(config: RenderSchedulerConfig): RenderScheduler {
  const scheduledLocalRenders = new ScheduledLocalRenders();
  const context = getChangeDetectorRefContext(config.cdRef);

  function schedule(): void {
    flush();

    const scheduledRender = animationFrameScheduler.schedule(() => {
      config.cdRef.detectChanges();
    });
    scheduledLocalRenders.set(context, scheduledRender);
  }

  function flush(): void {
    const scheduledRender = scheduledLocalRenders.get(context);
    
    if (scheduledRender) {
      scheduledRender.unsubscribe();
      scheduledLocalRenders.delete(context);
    }
  }

  return { schedule, flush };
}
```

`PushPipe`/`LetDirective`:

```ts
export class PushPipe implements OnDestroy {
  ngOnDestroy(): void {
    this.subscription.unsubscribe();
    this.renderScheduler.flush();
  }
}
```